internal/controllers/topology/cluster/patches/inline/json_patch_generator.go

```diff
@@ -167,6 +167,10 @@ func matchesSelector(req *runtimehooksv1.GeneratePatchesRequestItem, templateVar
 
 			// If the builtin variable could be read.
 			if err == nil {
+				// If templateMDClassJSON is defined, but no values are passed, consider it matches all
+				if selector.MatchResources.MachineDeploymentClass.Names != nil && len(selector.MatchResources.MachineDeploymentClass.Names) == 0 {
+					return true
+				}
```

Comment:
```suggestion
				// If templateMDClassJSON is defined and the selector is set to an empty object, consider it matches all.
				if len(selector.MatchResources.MachineDeploymentClass.Names) == 0 {
					return true
				}
```
nit: just a small simplification + go doc change

I think we don't really care about if the array is nil or not nil. We just want to check if it's empy in some way.

Reply:
Please add the following comment to the API type field ~ l.394 in clusterclass_types.go
```go
	// Note: If set to an empty object it will match all MachineDeploymentClasses.
```